lib/cucumber/core/test/result.rb

```diff
@@ -177,6 +177,24 @@ def ok?(be_strict = false)
           end
         end
 
+        class Ambiguous < Raisable
+          include Result.query_methods :ambiguous
+
+          def describe_to(visitor, *args)
+            visitor.failed(*args)
+            visitor.duration(duration, *args)
+            self
+          end
+
+          def to_s
+            "F"
+          end
+
+          def ok?(be_strict = false)
+            !be_strict
```

Comment:
But the feature file says [`Then it should fail with:`](https://github.com/MadameSheema/cucumber-ruby/blob/b39f24fdedaf39f67c0dc989bc2ee05548cbd2fa/features/docs/defining_steps/ambiguous_steps.feature#L36), and if it is not bad unless `--strict` then the scenario should say `Then it should pass with:`